INN/umbrella-currentorg
# Business Directory Plugin Template Override for Main Page

## Changes

This pull request makes the following changes:

- [x] customizes the main page template HTML output to show listing categories in the format described in #20.
- [x] styles that markup
- [x] Adds a Gruntfile and LESS source files to make styling the business Directory Plugin markup easier

<img width="1076" alt="Screen Shot 2019-06-25 at 18 35 23 " src="https://user-images.githubusercontent.com/1754187/60138493-94cf6400-9778-11e9-8deb-7bd0c59bc00f.png">

Note that in these images, the icons with the blue background are transparent; the blue background is applied to the image background via CSS. This is to make it easier to source icons for the various categories. 

I've included "Edit Listing Category" links for signed-in admin-level users here so that Current can more-easily modify the listings. They won't appear for non-admins.

On mobile it collapses to a one-column grid; there's an intermediary 2-column grid for browser windows between 600 and 900px wide. The upper limit on the width of the columns is 1170px from Largo's container element.

<img width="519" alt="Screen Shot 2019-06-25 at 18 43 03 " src="https://user-images.githubusercontent.com/1754187/60138678-3d7dc380-9779-11e9-821f-ddc80aa73708.png">

## Why

To match the client-provided mockups.
For #20

## Testing/Questions

Features that this PR affects:

- current.org/directory-of-services

Questions that need to be answered before merging:

- [ ] Do we know the full list of categories? This will make it easier to source icons.

Steps to test this PR:

1. Check out this branch
2. Visit the page with the `[businessdirectory]` shortcode, which for me is https://current.test/directory-of-services/
